compiler/noirc_evaluator/src/ssa.rs

```diff
@@ -141,6 +141,8 @@ pub(crate) fn optimize_into_acir(
         ssa.to_brillig(options.enable_brillig_logging)
     });
 
+    let ssa = ssa.inline_const_brillig_calls(&brillig);
```

Comment:
Shouldn't we do this earlier? The original issue mentioned doing this during SSA to take advantage of SSA optimizations but if we only do it at the very end of SSA there are no further optimizations.

Reply:
We need to do this relatively late in the compilation process as the unconstrained bytecode used is ideally the final bytecode.

Reply:
Not sure I understand - wouldn't doing it earlier result in more SSA optimizations? Even if doing it later would optimize the brillig part more so this pass has less to execute, we'd still prefer the SSA optimizations over that I'd think. Or am I missing something?

Reply:
Ok I think I mispoke with "need". It essentially boils down to two options. Comparing applying constant brillig function optimizations early in the compilation pipeline vs later, the pros/cons for doing this optimization early are (I may be missing some)

Pros:
- ACIR functions benefit more from the initial pass of SSA optimisations due to constant input brillig functions not blocking constant folding optimizations. (lesser need for additional passes)

Cons:
 - We end up needing to do another codegen of the brillig functions at the end to get optimized bytecode.
 - Brillig bytecode used for optimizations will end up being less efficient to execute so may slow down compilation.
   
   
The pros/cons of applying this optimization late are basically the reverse of the above.

Reply:
If we want 100% optimal bytecode then we need to do some iterations of constant folding and cfg_simplification anyway so I'm relatively relaxed about having some extra passes at the end (which we can easily turn off to improve compilation times).